### PR TITLE
Proper fix for LowCardinality together with JSONExtact functions

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This PR reverts https://github.com/ClickHouse/ClickHouse/pull/61617 and fixed the problem with usage of LowCardinality columns together with JSONExtract function. Previously the user may receive either incorrect result of a logical error.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit c49b2e458d14639c413bcf9919dd87f522a47fa0 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61957/c49b2e458d14639c413bcf9919dd87f522a47fa0/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Comment by jkartseva** [comment]
Looks like test failures happen because of removal
 ``` cpp
    DataTypePtr return_type;
    if (has_nothing_argument)
        return_type = std::make_shared<DataTypeNothing>();
    else if (null_presence.has_null_constant)
        return_type = makeNullable(std::make_shared<DataTypeNothing>());
    else if (null_presence.has_nullable)
        return_type = makeNullable(json_return_type);
    else
        return_type = json_return_type;
```

**Comment by nikitamikhaylov** [comment]
TL;DR We are removing LowCardinality from the requested type and then add it back after executing the extraction. 
Additionally functions now have their own implementation for the LowCardinality columns - just convert all of them to full ones.

**Review by jkartseva — APPROVED** [review]

File: src/Functions/FunctionsJSON.h [hunk 1]
```diff
@@ -348,17 +353,54 @@ class ExecutableFunctionJSON : public IExecutableFunction
...
+            ColumnsWithTypeAndName temporary_columns = null_presence.has_nullable ? createBlockWithNestedColumns(arguments) : arguments;
+            ColumnPtr temporary_result = chooseAndRunJSONParser(temporary_columns, json_return_type, input_rows_count);
+
+            if (null_presence.has_nullable)
+                temporary_result = wrapInNullable(temporary_result, arguments, result_type, input_rows_count);
+
+            if (result_type->lowCardinality())
+                temporary_result = recursiveLowCardinalityTypeConversion(temporary_result, json_return_type, result_type);
+
+            return temporary_result;
```
[the remark this answers is not in the document]
Comment (nikitamikhaylov) [reply src/Functions/FunctionsJSON.h:413]: The code with lambda looks dirtier...
